modules/cloudsql-postgres/main.tf

```diff
@@ -0,0 +1,122 @@
+terraform {
+  required_providers {
+    google = {
+      source = "hashicorp/google"
+    }
+  }
+}
+
+# Labels
+
+locals {
+  default_labels = {
+    cloudsql = var.name
+  }
+
+  squad_label = var.squad != "" ? {
+    squad = var.squad
+    team  = var.squad
+  } : {}
+
+  merged_labels = merge(local.default_labels, local.squad_label)
+}
+
+# Primary Cloud SQL instance
+
+resource "google_sql_database_instance" "this" {
+  name    = var.name
+  project = var.project
+  region  = var.region
+
+  database_version    = var.database_version
+  deletion_protection = var.deletion_protection
+
+  settings {
+    tier                  = var.tier
```

Comment:
should we add an option for `edition`
https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/sql_database_instance#edition-1

Reply:
I think we should because either way Google provider will choose a default, and it's nicer to be explicit.

> Note: If the database version for your instance is PostgreSQL 16 or later, then the default Cloud SQL for PostgreSQL edition is Cloud SQL Enterprise Plus edition. If the database version is earlier than PostgreSQL 16, then the default edition is Cloud SQL Enterprise edition.

https://cloud.google.com/sql/docs/postgres/create-instance